File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/Manacher.cpp
#include <string>
#include <vector>

class Manacher {
public:
    Manacher(const std::string& input_string) : input_string(input_string) {}

    std::string palindromic_string();

private:
    int palindromic_length(int center, int diff, const std::string& s) {
        if (center - diff < 0 || center + diff >= s.length()) {
            return 0;
        }
        if (s[center-diff] != s[center+diff]) {
            return 0;
        }
        return 1 + palindromic_length(center, diff+1, s);
    }

    std::string input_string;
};

std::string Manacher::palindromic_string() {
    if (input_string.empty()) {
        return "";
    }
    if (input_string.length() == 1) {
        return input_string;
    }

    std::string new_input_string;
    for (int i = 0; i < input_string.length() - 1; i++) {
        new_input_string += input_string[i];
        new_input_string += '|';
    }
    new_input_string += input_string.back();

    int max_length = 0;
    int start = 0;

    for (int i = 0; i < new_input_string.length(); i++) {
        int length = palindromic_length(i, 1, new_input_string);
        if (max_length < length) {
            max_length = length;
            start = i;
        }
    }

    std::string substring = new_input_string.substr(start - max_length, 2*max_length+1);
    std::string output_string;
    for (char c : substring) {
        if (c != '|') {
            output_string += c;
        }
    }
    return output_string;
}

File: data/derived_cot/rq1_traces/translation/glm5.1/cpp_to_py/code_output/Manacher.py
class Manacher:
    def __init__(self, input_string):
        self.input_string = input_string

    def palindromic_length(self, center, diff, s):
        if (center - diff == -1) or (center + diff == len(s)) or (s[center - diff] != s[center + diff]):
            return 0
        return 1 + self.palindromic_length(center, diff + 1, s)

    def palindromic_string(self):
        max_length = 0
        new_input_string = ""
        output_string = ""

        for i in range(len(self.input_string) - 1):
            new_input_string += self.input_string[i] + "|"
        new_input_string += self.input_string[-1]

        start = 0
        for i in range(len(new_input_string)):
            length = self.palindromic_length(i, 1, new_input_string)

            if max_length < length:
                max_length = length
                start = i

        for i in range(start - max_length, start + max_length + 1):
            if new_input_string[i] != '|':
                output_string += new_input_string[i]

        return output_string